bigbluebutton-html5/imports/ui/components/user-avatar/component.jsx

```diff
@@ -43,7 +43,7 @@ const UserAvatar = ({
       [styles.presenter]: presenter,
       [styles.muted]: muted,
       [styles.listenOnly]: listenOnly,
-      [styles.talking]: talking,
+      [styles.talking]: (talking && !muted),
```

Comment:
This is adding business logic to a presentation component, you should avoid that.

Reply:
I'd like to add my 5 cents here.

Business logic is "how things work", for example a container can make a decision whether to display one component or another.
In this case the ring around the avatar is the CSS class, a part of this particular component. And this component is supposed to make a decision on how it should look based on the props provided. No matter if that's 2 properties, or just 1 (`shouldDisplayBorder`).

Swapping the CSS class IS NOT business logic.

Reply:
Then in everyplace we want to check if the user is talking, the developer should be adding the `&& muted` check?

Like here as an example
https://github.com/bigbluebutton/bigbluebutton/blob/master/bigbluebutton-html5/imports/ui/components/actions-bar/mute-button/component.jsx#L29

:wink:

Reply:
Having it as one property would be more convenient, of course.
But the check itself won't go anywhere from the component, it will still have to check:
```
if (isTalking) {
    className = styles.circleGlow;
}
```
But it would be checking 1 property instead of 2 or 3.